presentation/components/src/presentation-components/properties/NavigationPropertyTargetSelector.tsx

```diff
@@ -9,37 +9,41 @@ import {
   components, ControlProps, IndicatorContainerProps, IndicatorProps, MenuProps, OptionProps, OptionTypeBase, ValueContainerProps,
 } from "react-select";
 import { AsyncPaginate } from "react-select-async-paginate";
-import { PropertyRecord, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
-import { PropertyEditorProps } from "@itwin/components-react";
+import { PropertyDescription, PropertyRecord, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { PropertyEditorProps, PropertyValueRendererManager } from "@itwin/components-react";
 import { IModelConnection } from "@itwin/core-frontend";
 import { SvgCaretDownSmall } from "@itwin/itwinui-icons-react";
-import { NavigationPropertyInfo } from "@itwin/presentation-common";
+import { InstanceKey, LabelDefinition, NavigationPropertyInfo } from "@itwin/presentation-common";
 import { translate } from "../common/Utils";
 import {
   NavigationPropertyTarget, useNavigationPropertyTargetsLoader, useNavigationPropertyTargetsRuleset,
 } from "./UseNavigationPropertyTargetsLoader";
 
+/** @internal */
 export interface NavigationPropertyTargetSelectorAttributes {
   getValue: () => PropertyValue | undefined;
   divElement: HTMLDivElement | null;
 }
 
+/** @internal */
 export interface NavigationPropertyTargetSelectorProps extends PropertyEditorProps {
   imodel: IModelConnection;
-  getNavigationPropertyInfo: (record: PropertyRecord) => Promise<NavigationPropertyInfo | undefined>;
+  getNavigationPropertyInfo: (property: PropertyDescription) => Promise<NavigationPropertyInfo | undefined>;
+  propertyRecord: PropertyRecord;
 }
 
+/** @internal */
 export const NavigationPropertyTargetSelector = React.forwardRef<NavigationPropertyTargetSelectorAttributes, NavigationPropertyTargetSelectorProps>((props, ref) => {
   const { imodel, getNavigationPropertyInfo, propertyRecord, onCommit, setFocus } = props;
   const divRef = React.useRef<HTMLDivElement>(null);
-  const targetsRuleset = useNavigationPropertyTargetsRuleset(getNavigationPropertyInfo, propertyRecord);
+  const targetsRuleset = useNavigationPropertyTargetsRuleset(getNavigationPropertyInfo, propertyRecord.property);
   const loadTargets = useNavigationPropertyTargetsLoader({ imodel, ruleset: targetsRuleset });
 
-  const [selectedTarget, setSelectedTarget] = React.useState<NavigationPropertyTarget | undefined>();
+  const [selectedTarget, setSelectedTarget] = React.useState(() => getNavigationTargetFromPropertyRecord(propertyRecord));
 
   const onChange = React.useCallback((target?: NavigationPropertyTarget) => {
     setSelectedTarget(target);
-    target && propertyRecord && onCommit && onCommit({ propertyRecord, newValue: getPropertyValue(target) });
+    target && onCommit && onCommit({ propertyRecord, newValue: getPropertyValue(target) });
```

Comment:
Looks like we're only going to call `onCommit` if there is a selected value. What about navigation properties that have an optional target, can we handle those?

Reply:
It does not allow to select empty target at the moment. I filled an issue to add support for optional targets #4433